token-lending/program/src/oracles.rs

```diff
@@ -0,0 +1,348 @@
+#![allow(missing_docs)]
+use crate::{
+    self as solend_program,
+    error::LendingError,
+    math::{Decimal, TryDiv, TryMul},
+};
+use pyth_sdk_solana;
+use solana_program::{
+    account_info::AccountInfo, msg, program_error::ProgramError, sysvar::clock::Clock,
+};
+use std::{convert::TryInto, result::Result};
+
+pub fn get_pyth_price(
+    pyth_price_info: &AccountInfo,
+    clock: &Clock,
+) -> Result<Decimal, ProgramError> {
+    const MAX_PYTH_CONFIDENCE_RATIO: u64 = 10;
+    const STALE_AFTER_SECONDS_ELAPSED: u64 = 120;
+
+    if *pyth_price_info.key == solend_program::NULL_PUBKEY {
+        return Err(LendingError::NullOracleConfig.into());
+    }
+
+    let price_feed = pyth_sdk_solana::load_price_feed_from_account_info(pyth_price_info)?;
+    let pyth_price = price_feed
+        .get_latest_available_price_within_duration(
```

Comment:
think it's better to use slots to find prev prices instead of timestamps, since solana timestamps have been known to drift

Reply:
can copy logic for this: https://docs.rs/pyth-sdk/0.6.1/src/pyth_sdk/lib.rs.html#297

Reply:
actually after talking to pyth, this seems like a non-issue